boards/shared/Src/Io/_Io_SharedAdc.c

```diff
@@ -0,0 +1,121 @@
+#include <stdint.h>
+#include <assert.h>
+#include "_Io_SharedAdc.h"
+#include "App_SharedConstants.h"
+
+struct AdcInput
+{
+    ADC_HandleTypeDef *hadc;
+    size_t             vrefint_index;
+    uint32_t           num_active_channels;
+    uint16_t           adc_max_value;
+    uint16_t *         adc_readings;
+};
+
+/**
+ * Gets the ADC resolution for the given hadc
+ * @param hadc The ADC handle for the ADC we are initializing
+ * @return adc_max_resolution The maximum ADC resolution for the given ADC
+ * handle
+ */
+static uint16_t _Io_GetAdcResolution(ADC_HandleTypeDef *hadc);
+
+static uint16_t _Io_GetAdcResolution(ADC_HandleTypeDef *hadc)
+{
```

Comment:
This is the original implementation. I'm wondering if I should return an ErrorStatus for this implementation :thinking: :thailand: 

```
static ErrorStatus InitializeAdcMaxValue(ADC_HandleTypeDef *hadc)
{
    switch (hadc->Init.Resolution)
    {
        case ADC_RESOLUTION_6B:
            adc_max_value = MAX_6_BITS_VALUE;
            break;
        case ADC_RESOLUTION_8B:
            adc_max_value = MAX_8_BITS_VALUE;
            break;
        case ADC_RESOLUTION_10B:
            adc_max_value = MAX_10_BITS_VALUE;
            break;
        case ADC_RESOLUTION_12B:
            adc_max_value = MAX_12_BITS_VALUE;
            break;
        default:
            return ERROR;
    }
    return SUCCESS;
}
```